dao/src/main/java/org/wso2/testgrid/dao/repository/TestPlanRepository.java

```diff
@@ -225,4 +225,22 @@ public TestPlan getLastFailure(TestPlan testPlan) {
             return null;
         }
     }
+
+    /**
+     * This method returns all the test plans that belongs to same infrastructure set,same deployment pattern
+     * and same product.
+     *
+     * @param testPlan testPlan being queried
+     * @return a List of {@link TestPlan} representing the history of that test plan
+     */
+    public List<TestPlan> getTestPlanHistory(TestPlan testPlan) {
+        String sql = " select t.* from test_plan t inner join deployment_pattern dp inner " +
+                "join product p on p.id=dp.PRODUCT_id and dp.id=t.DEPLOYMENTPATTERN_id " +
+                "where t.infra_parameters=? AND dp.id=? AND p.id=?";
```

Comment:
We can easily write these queries using JPA Criteria API instead of using JPQL native queries. Native queries lacks type-safety, and has lots of other drawbacks. See [1].

It's ok to have it for this PR.. :)

[1] https://github.com/wso2/testgrid/issues/413

Reply:
Will fix in the next milestone